packages/check-html-links/src/checkLinks.js

```diff
@@ -0,0 +1,59 @@
+import fetch from 'node-fetch';
+
+/**
+ * @type {Map<string,boolean>}
+ */
+const resultsMap = new Map();
+
+/**
+ *
+ * @param {string} url
+ * @param {boolean} result
+ * @returns {boolean}
+ */
+const memorizeCheckup = (url, result) => {
+  resultsMap.set(url, result);
+  return result;
+};
+
+/**
+ *
+ * @param {string} url
+ * @param {string} method
+ * @returns
+ */
+const fetchWrap = async (url, method = 'GET') =>
+  fetch(url, { method })
+    .then(response => response.ok)
+    .catch(() => false);
+
+/**
+ *
+ * @param {string} url
+ * @returns {Promise<boolean>}
+ */
+const fetchHead = async url => fetchWrap(url, 'HEAD');
+
+/**
+ *
+ * @param {string} url - URL object to check
+ * @returns {Promise<boolean>} true if url is alive or false if not
+ */
+const checkUrl = async url =>
+  (fetchHead(url) || fetchWrap(url)).then(result => memorizeCheckup(url, result));
```

Comment:
@daKmoR @jorgecasar I just played around with it and ran it against my built site. 

The command checking external links never succeeds because it hangs at one URL (`https://twitter.com/intent/tweet?text='HTTP%20headers%20for%20the%20responsible%20developer'&url=https://www.stefanjudis.com/blog/http-headers-for-the-responsible-developer/&via=stefanjudis`). 

Not really sure what's up, but a timeout option might be necessary. :)

Reply:
I implemented a time out and this indeed did the trick! 

```js
const fetchWrap = async (url, method = 'GET') => {
  return Promise.race([
    fetch(url, { method })
      .then(response => response.ok)
      .catch(() => false),
    new Promise(resolve => setTimeout(resolve, 10000, false)),
  ]);
};
```

A few observations:

- redirects such as `https://amzn.to/3rF9gfa` a considered as not okay. I'm not sure if it has to be that strict.
- `node-fetch` def has a problem with the mentioned Twitter URL

But it found a bunch of dead links and only needed `Checking links duration: 941s 272.128865ms`. 🎉